### ISSUE TITLE
Grouped form controls missing an accessible name - Siteimprove - SiteNow Intranet Form API Fieldsets

### ISSUE THREAD

**Issue by joewhitsitt** [uiowa/uiowa#9497]
From https://github.com/uiowa/uiowa/issues/9462

This differs from https://github.com/uiowa/uiowa/issues/9293 in that this issue is related to `'#type' => 'fieldset',` that are currently missing the `#title` property.

The description property does get linked up (aria-describedby) but is not a replacement for an accessible name.

# Proposed Solution

- Add `#title` to these Form API implementations.
- https://github.com/uiowa/uiowa/blob/5766b73628fbf4a471c239891995d50a9ef9b0c2/docroot/modules/custom/sitenow_intranet/src/Form/SettingsForm.php#L57
- https://github.com/uiowa/uiowa/blob/5766b73628fbf4a471c239891995d50a9ef9b0c2/docroot/modules/custom/sitenow_intranet/src/Form/SettingsForm.php#L35
